### ISSUE TITLE
Grouping breaks Column Header Height for deferred columnDefs generation

### ISSUE THREAD

**Issue by JLLeitschuh** [angular-ui/ng-grid#3609]
http://plnkr.co/edit/u6LBOx?p=preview
As you can see in this demo if the column header height is wrong when the grouping directive is added to the grid. This is only a problem when the column defs aren't there when the grid options are loaded.
Example:
```
$timeout(function(){
  var newColumnDefs = [
    { name:'firstName', field: 'first-name', displayName: 'First Name'},
    { name:'1stFriend', field: 'friends[0]', displayName: '1st Friend'},
    { name:'city', field: 'address.city', displayName: 'City'},
    { name:'getZip', field: 'getZip()'}
  ];
  $scope.gridOptions.columnDefs = newColumnDefs;
}, 500);
```
This problem is not there for the grid without the grouping directive.

When you perform the grouping action the headers appear correctly however when you ungroup they return to their broken state.
OS: Mac OSX
Browser: Chrome
Build: v3.0.0-rc.21-d68dcc6 - 2015-05-27
![screen shot 2015-05-27 at 2 19 41 pm](https://cloud.githubusercontent.com/assets/1323708/7844094/01c155bc-047c-11e5-9b51-37c44c1160ae.png)

**Comment by AgDude** [comment]
I don't think this is specifically related to grouping. This has to do with multiple render containers and no columns. This may be due to my refactor a few weeks ago of the header height calculation, but I am not seeing why.  @c0bra would you mind taking a look?

The original plunker doesn't apply ```notifyDataChange``` so here is a fork that does, and the problem still occurs: http://plnkr.co/edit/implXX?p=preview

**Comment by c0bra** [comment]
Hmm. Not sure, I will take a look. Is this the commit @AgDude? 65ad61f4439983179be4d44bac3a6cc9151ac094

- [ ] test grouping with delayed columnDefs before and after header height refactor

**Comment by AgDude** [comment]
Yes, that is the commit I am referring to. I haven't tried replicating around the time of the commit.

**Comment by JLLeitschuh** [comment]
Still a bug in 3.0
